### PR TITLE
DROOLS-4882 : [Guided DecisionTable] Incorrect getters and setters for fields that starts with one lowercase letter

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.redhat.com/browse/DROOLS-4882

### CODE REVIEW

**Review by manstis — APPROVED** [review]
LGTM.

**Comment by jomarko** [comment]
@Rikkola Hello, I am sorry, I am confused with your message "cherry picked from". I expect such message in cases we have the issue already solved in some stream, but neither DROOLS-4882 or RHPAM-412 seems to be resolved. Could you please explain me what are we cherry-picking?

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 1]
```diff
@@ -1735,6 +1735,14 @@ protected void generateModifyMethodSeparator(final RHSGeneratorContext gctx,
             }
         }
 
+        private char formatFieldFirstCharacterToFitDroolsCoreStandards(final String fieldName) {
+            if (fieldName.length() > 2 && Character.isLowerCase(fieldName.charAt(0)) && Character.isUpperCase(fieldName.charAt(1))) {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java]: I think the condition needs top start as `fieldName.length() > 1` instead  of `fieldName.length() > 2 `.

I think the generated code getters/setters has same rules for fields **aField** and **aF**. Or am I wrong?

**Review by jomarko — CHANGES_REQUESTED** [review]
I think we need to update the PR due to a corner case - two letter fields in the form [lower case letter][upper case letter].

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java (hunk 1 above)
Comment (Rikkola) [reply drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java]: Fixed and unit test updated to cover this.

**Comment by Rikkola** [comment]
@jomarko The "cherry picked from" was due to me cherry-picking this from another branch that had other changes not related to this ticket. I removed the text from the commit. Ready for rereview now.

**Review by jomarko — APPROVED** [review]
Thank you @Rikkola

**Comment by Rikkola** [comment]
Jenkins please execute SonarCloud Code Analysis

**Comment by Rikkola** [comment]
Jenkins please execute SonarCloud Code Analysis

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can't get the Analysis green due to it being broken. Based on the repository commit history this does not stop merges. So can I have a merge please :)

**Pull request merged**